### PR TITLE
Ethernet statistics support

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: subsys/net/ip/Kconfig.stats [hunk 1]
```diff
@@ -93,4 +93,13 @@ config NET_STATISTICS_MLD
 	help
 	  Keep track of MLD related statistics
 
+config NET_STATISTICS_ETHERNET
+	bool "Ethernet statistics"
+	depends on NET_L2_ETHERNET
+	default y
```
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/ip/Kconfig.stats]: Well, if you have statistics and ethernet enabled, then I would guess that you are also interested in ethernet statistics. But this is of course a thing we can debate on :)

File: subsys/net/ip/net_stats.c [hunk 2]
```diff
@@ -323,6 +324,20 @@ static int net_stats_get(u32_t mgmt_request, struct net_if *iface,
...
+
+		if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
+			return -ENOENT;
+		}
+
+		eth = net_if_get_device(iface)->driver_api;
+		len_chk = sizeof(struct net_stats_eth);
+		src = eth->stats;
+		break;
+	}
```
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/ip/net_stats.c]: This sounds very counter intuitive as the ethernet_mgmt.c contains only management stuff and has nothing to do with statistics.

File: subsys/net/ip/l2/eth_stats.h [hunk 3]
```diff
@@ -0,0 +1,135 @@
+/*
```
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/ip/l2/eth_stats.h]: Sure

File: drivers/ethernet/eth_native_posix.c [hunk 4]
```diff
@@ -86,12 +91,26 @@ static int eth_send(struct net_if *iface, struct net_pkt *pkt)
...
 	}
 
-	net_pkt_unref(pkt);
+	eth_stats_update_bytes_tx(iface, count);
+	eth_stats_update_pkts_tx(iface);
+
+	if (IS_ENABLED(CONFIG_NET_STATISTICS_ETHERNET)) {
+		if (net_eth_is_addr_broadcast(
+			    &((struct net_eth_hdr *)NET_ETH_HDR(pkt))->dst)) {
+			eth_stats_update_broadcast_tx(iface);
```
Comment (pfl) [inline drivers/ethernet/eth_native_posix.c]: Isn't eth_stats_update_broadcast_tx() defined as nothing in the previous patch when CONFIG_NET_STATISTICS_ETHERNET is disabled? Can we in that case do without the IS_EANBLED macro and let the compiler mop up stuff? Same for the rest below.

File: subsys/net/ip/net_stats.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/ip/net_stats.c]: Lets have a compromise and put it in ethernet_stats.c instead.

[the remark this answers is not in the document]
Comment (tbursztyka) [reply subsys/net/ip/net_stats.c]: yep

File: drivers/ethernet/eth_native_posix.c (hunk 4 above)
Comment (jukkar) [reply drivers/ethernet/eth_native_posix.c]: In theory we could try to optimize this more, but I do not see much point as IS_ENABLED() will remove the block nicely.

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**